### ISSUE TITLE
`unused_self` unsexpected positive with `#![allow(unused)]`

### ISSUE THREAD

**Issue by tgross35** [rust-lang/rust-clippy#10117]
### Summary

When `allow(unused)` is enabled (e.g. during API development), it is surprising that the `unused_self` lint throws an error - especially when there is only a `todo!()` in the body. I would expect either that allowing `unused` disables the `unused_self` warning, or that the presence of `todo` should ignore it (to be detected by the `todo` lint)

### Lint Name

unused_self

### Reproducer

I tried this code [playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=f077d55fa1f1549942b98a387f2aa53f):

```rust
#![allow(unused)]
#![warn(clippy::unused_self)]

struct X;

impl X {
    pub fn test_fn(&self) {
        todo!();
    }
}
```

I saw this happen:

```
warning: unused `self` argument
 --> src/lib.rs:7:20
  |
7 |     pub fn test_fn(&self) {
  |                    ^^^^^
  |
  = help: consider refactoring to a associated function
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_self
note: the lint level is defined here
 --> src/lib.rs:2:9
  |
2 | #![warn(clippy::unused_self)]
  |         ^^^^^^^^^^^^^^^^^^^

warning: `playground` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.39s
```

I expected to see this happen: no lint

### Version

```text
0.1.68 nightly (playground)
```
